src/backend/gpopt/translate/CTranslatorDXLToPlStmt.cpp

```diff
@@ -3763,12 +3763,6 @@ CTranslatorDXLToPlStmt::GetFlowCTEConsumer(List *shared_scan_cte_consumer_list)
 		}
 	}
 
-	if (NULL == flow)
-	{
-		flow = MakeNode(Flow);
-		flow->flotype = FLOW_UNDEFINED;	 // default flow
-	}
```

Comment:
> this function may return null values that are used as flow for descendant plan nodes (Sort or Material). Why assignment of nulls is correct?

because only flow under Motion node is significant, not under shared scan

Reply:
I have another question about this block of code. Will it be naive to keep it and remove only one FLOW_SINGLETON assertion? By this, we can make patch less invasive.

Reply:
> remove only one FLOW_SINGLETON assertion

was first idea

Reply:
So what is the reason of choosing a current option?

Reply:
> So what is the reason of choosing a current option?

firstly, to not incorrectly fill flow by orca
and secondly, to to ensure that flow for node under Motion is correctly filled and on coordinator it has flotype = FLOW_SINGLETON and segindex = -1

Reply:
I removed this function, but saved assert.

Reply:
To summarize: the `flow` of `ShareInputScan/Material/Sort` nodes are not significant to make decision at which segments (segment(s) or coordinator) the slice will be executed.
So now `flow` of `Material` and `Sort` nodes are not filled and for `ShareInputScan` nodes there is an assertion which checks that `ShareInputScan` nodes (where `flow` is not NULL) must have the same `flotype`. 
I'm right?

Reply:
> To summarize: the `flow` of `ShareInputScan/Material/Sort` nodes are not significant to make decision at which segments (segment(s) or coordinator) the slice will be executed. So now `flow` of `Material` and `Sort` nodes are not filled and for `ShareInputScan` nodes there is an assertion which checks that `ShareInputScan` nodes (where `flow` is not NULL) must have the same `flotype`. I'm right?

Yes